project/src/main/webapp/gmail.js

```diff
@@ -0,0 +1,205 @@
+      *  Initializes the API client library and sets up sign-in state
+      *  listeners.
+      */
+    function initClient() {
+        gapi.client.init({
+            apiKey: API_KEY,
+            clientId: CLIENT_ID,
+            discoveryDocs: DISCOVERY_DOCS,
+            scope: SCOPES
+        }).then(function () {
+            // Listen for sign-in state changes.
+            gapi.auth2.getAuthInstance().isSignedIn.listen(updateSigninStatus);
+
+            // Handle the initial sign-in state.
+            updateSigninStatus(gapi.auth2.getAuthInstance().isSignedIn.get());
+            authorizeButton.onclick = handleAuthClick;
+            signoutButton.onclick = handleSignoutClick;
+        },  function(error) {
+            appendPre(JSON.stringify(error, null, 2));
+        });
+    }
+
+    /**
+      *  Called when the signed in status changes, to update the UI
+      *  appropriately. After a sign-in, the API is called.
+      */
+    function updateSigninStatus(isSignedIn) {
+        if (isSignedIn) {
+            authorizeButton.style.display = 'none';
+            signoutButton.style.display = 'block';
+            // getMessage();
+            listMessages();
+        }   else {
+            authorizeButton.style.display = 'block';
+            signoutButton.style.display = 'none';
+        }
+    }
+
+    /**
+      *  Sign in the user upon button click.
+      */
+    function handleAuthClick(event) {
+        gapi.auth2.getAuthInstance().signIn();
+    }
+
+    /**
+      *  Sign out the user upon button click.
+      */
+    function handleSignoutClick(event) {
+        gapi.auth2.getAuthInstance().signOut();
+    }
+
+    /**
+      * Append a pre element to the body containing the given message
+      * as its text node. Used to display the results of the API call.
+      *
+      * @param {string} message Text to be placed in pre element.
+      */
+    function appendPre(message) {
+        var pre = document.getElementById('content');
+        var textContent = document.createTextNode(message + '\n');
+        pre.appendChild(textContent);
+    }
+
+
+    async function keywordExtraction(userInput) {
+    var result = "";
+    var count = 0;
+    var searchQuery = userInput.split(" ");
+    for (word of searchQuery){
+        if ((searchQuery.length-1) == count && word.indexOf('?') != -1) {
+                word = word.substring(0, word.length-1);
+            }
+        var url = new URL("https://api.datamuse.com/words?md=p&sp=" + word + "&max=1");
+        const output = await fetch(url).then(response => response.text());
+        if (output.includes("[\"n\"]") == true || output.includes("[\"adj\"]") == true){
+            result += word;
+            if (count != (searchQuery.length-1)){
+                result += " ";
+            }
+        }
+        count++;
+    }
+    return result; 
+    }
+    
+    //Retrieve messages using hardcoded queries and the signed-in email.
+    async function listMessages() {
```

Comment:
Like other comment try to break down for easy readability. A lot of nested statements in function